### ISSUE TITLE
Drop In crashes/closes with no apparent error when selecting Google Pay.

### ISSUE THREAD

**Issue by soarb** [braintree/braintree-android-drop-in#355]
### General information

* SDK/Library version: com.braintreepayments.api:drop-in:6.2.0 
* Environment: Sandbox 
* Android Version and Device: Samsung Galaxy S10 with Android 12.0

### Issue description

I'm in the process of migrating our Android Drop-In integration to v6 and can so far successfully use the DropInClient with my custom ClientTokenProvider to retrieve the last used payment method.

I've since moved on to presenting the Drop In UI to manage payment methods etc. All aspects of the UI appear to work fine apart from when it comes to selecting Google Pay. In this event, the Drop In UI just disappears, none of my DropInListener events are called and I can't see anything obvious in Logcat at the time of tapping the Google Pay option to indicate any errors.

I use the following constructor to create my DropInClient:-
```
public DropInClient(FragmentActivity activity, 
  DropInRequest dropInRequest, ClientTokenProvider clientTokenProvider) ...
```

and in call cases this appears to work except when selecting Google Pay.

However, if I configure a new DropInClient using the deprecated constructor:-
```
public DropInClient(Context context, String authorization, DropInRequest dropInRequest) ...
```

and then call `launchDropInForResult`, again the Drop In UI shows, and tapping the Google Pay option works fine.

Am I missing something in the docs or the migration guide as I'd like to avoid using any of these deprecated method?

Cheers

Ben

**Comment by soarb** [comment]
Thanks for the quick reply @sshropshire :) .. yes I can confirm that I'm calling setListener and implementing the DropInListener interface. These methods _are_ called for any interaction I perform within the drop-in, but for some reason, tapping Google Pay just kills the drop-in and I receive nothing back ... no response, no error - it's pretty weird :(

**Comment by sshropshire** [comment]
@soarb interesting yeah. Have you tried running our sample app and verifying the same behavior? Or alternatively if you could implement a sample app where we can reproduce the issue that would help us a lot to resolve the issue.

**Comment by soarb** [comment]
@sshropshire I'll do my best :) ... am on a deadline to get this working, so I'll fallback on what I know works first, i.e., the deprecated constructor and launchDropInForResult method, then try and get you an example app if that's OK?

**Comment by sshropshire** [comment]
@soarb yeah definitely. The primary motivation behind us deprecating those constructors is to start leveraging the newly recommended [Activity Result APIs](https://developer.android.com/training/basics/intents/result) since `startActivityForResult()` is now deprecated.

**Comment by soarb** [comment]
OK some small progress @sshropshire :)

I must confess that I'm writing a react-native module to support both the Braintree SDK and iOS SDK in our App ... and this has been working for years on the 3.x SDK and the 5.x drop-in for Android.

I think my main problem is a very basic misunderstanding of how to use the DropInClient.

I was initialising the client with an empty DropInRequest and keeping a reference to it.

When my app wanted to interact with the DropInClient I'd reset the reference to a new DropInRequest instance and configure it for my use case e.g., just fetch the last used payment method (new DropInRequest()), or display the full ui and enable vault management whilst supplying a GooglePayRequest etc.

Rather than resetting the reference to a new instance, I just updated the existing reference with the new props et volia ... tapping the Google Pay button worked!

I apologise for this being a bit wordy, but I feel like I've completely misunderstood how to use the DropInClient.

Rather than initialising it all in my application's main Activity and keeping references around for future use, should I be spawning other Activity instances on demand e.g., for showing the drop-in ui I should create an Intent and start a new activity just for that purpose and configure the DropInClient in _that_ Activity's onCreate method?

Any pointers would be much appreciated if you can spare the time!

Cheers :)

**Comment by sshropshire** [comment]
@soarb all good! We welcome all feedback it helps us to make the SDK better.

The main constraint at the moment for `DropInClient` is it needs to be instantiated in the `onCreate()` method of an `Activity` or a `Fragment`.

The best workaround for now is as you've mentioned to keep a reference to the `DropInRequest` used to create the `DropInClient`. You can then modify the request before calling `launchDropIn()`.

Internally, we have discussed removing the requirement to have a `DropInRequest` when creating a `DropInClient` – we're working on prioritization at the moment. Would that help in this scenario?

**Comment by soarb** [comment]
@sshropshire To my mind at least, it would definitely make more sense to remove the requirement to have a DropInRequest when creating a DropInclient - it then feels more consistent with the ThreeDSecureClient too.

Having said all this, I've finally hit upon a solution that's working well for my react-native module for Android (again, I think the key thing here was my inexperience with Android Activity instances etc).

1. The main module when react-native registers it, just creates a DropInClient (with token provider) with an empty DropInRequest which serves the purpose of being able to query a customer's last known payment method (i.e., no UI requirement).
2. For presenting the DropIn UI via DropInClient, Google Payment Sheet (GooglePayClient) and performing 3DS 2.0 flow using the ThreeDSecureClient which (potentially) needs to to issue a challenge to the user via browser switching, I've created my own additional activities to start each of these processes. As these Activity instances always spawn a new screen, the trick was to ensure they were completely transparent and the App was still visible beneath whilst whichever client I needed to use presented the necessary UI

I can now happily report that so far, everything I've thrown at my App when pointed at our Sandbox account is working very well indeed!

I think I can definitely close this issue for now :) and again, thank you for all your support!

Cheers

Ben
